plugins/BEdita/Core/src/Model/Table/UsersTable.php

```diff
@@ -122,11 +122,17 @@ public function validationSignup(Validator $validator)
 
         $validator
             ->email('email')
-            ->requirePresence('email')
-            ->add('email', 'unique', ['rule' => 'validateUnique', 'provider' => 'table'])
+            ->add('email', 'unique', ['rule' => 'validateUnique', 'provider' => 'table']);
 
-            ->requirePresence('password_hash')
-            ->notEmpty('password_hash');
+        if (Configure::read('Signup.requirePassword', true)) {
+            $validator
+                ->requirePresence('password_hash')
+                ->notEmpty('password_hash');
+        }
+
+        if (Configure::read('Signup.requireEmail', true)) {
+            $validator->requirePresence('email');
```

Comment:
(I'm leaving this comment here, but it applies to several lines)

Using this approach, the configuration key is read at the time the validator is built. This is probably fine in most cases, but makes it impossible to change the configuration at runtime in a reliable and predictable way.

I'd prefer another approach that evaluates the requirements at the time the data is being validated:

```php
$validator->requirePresence('email', function () {
    return Configure::read('Signup.requireEmail', true);
});
```